### PR TITLE
Feature/4621 put supplier

### PR DESCRIPTION
**Pull request by RuthBerry96**
Note: This is without validation

Story: [AB#4621](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4621)
Task: [AB#7181](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7181)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/SupplierSectionController.cs [hunk 1]
```diff
@@ -50,18 +51,58 @@ public async Task<ActionResult<SupplierModel>> GetAsync(string orderId)
...
 
+            order.SupplierId = model.SupplierId;
+            order.SupplierName = model.Name;
+
+            var address = model.Address;
+
+            if (order.SupplierAddress is null)
+                order.SupplierAddress = new Address();
+
+            order.SupplierAddress.Line1 = address.Line1;
```
Comment (CallumBradburyBJSS) [inline src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/SupplierSectionController.cs:82]: It'd be cool if we had a function on Address, or an extension method on Address, to do this for us, but isn't essential

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    Given Orders exist
+        | OrderId    | OrganisationId                       | OrganisationName | OrganisationOdsCode | SupplierId | SupplierAddressPostcode | SupplierContactEmail    | Description      | OrderStatusId | Created    | LastUpdated | LastUpdatedBy                        |
+        | C000014-01 | 4af62b99-638c-4247-875e-965239cd0c48 | Hampshire CC     | 432432              | SupId      | LS 1 3AP                | Fred.robinson@email.com | Some Description | 1             | 11/05/2020 | 11/05/2020  | 335392e4-4bb1-413b-9de5-36a85c9c0422 |
+    And the user is logged in with the Buyer role for organisation 4af62b99-638c-4247-875e-965239cd0c48
+
+@4621
+Scenario: 1. Update a supplier section
+    Given the user wants to update the SupplierAddress section for the address
+        | Line1     | Line2      | Line3       | Line4          | Line5          | Town         | County  | Postcode | Country        |
+        | New Line1 | Lower Flat | Rocks Close | Larger Village | Masive Village | Another Town | N Yorks | YO11 1AP | United Kingdom |
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature:22]: if we're being cool and saying Massive Village, let's not call it Masive Village ;)

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/ContactSteps.cs [hunk 3]
```diff
@@ -65,6 +65,13 @@ public async Task ThenTheContactPrimaryContactIsReturned(string section, Table t
             actual.Should().BeEquivalentTo(expected);
         }
 
+        [Given(@"the user wants to update the (.*) section for the contact")]
+        public void WhenTheUserWantsToUpdateTheSupplierContactSectionForTheContact(string section, Table table)
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/ContactSteps.cs:69]: Remove Supplier from this function name

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs [hunk 4]
```diff
@@ -86,16 +90,75 @@ public async Task ThenTheResponseContainsTheFollowingSupplierAddress(Table table
...
 
             var expected = table.CreateInstance<SupplierAddressTable>();
             actual.Should().BeEquivalentTo(expected);
         }
 
+        [When(@"the user makes a request to update the supplier with order ID (.*)")]
+        public async Task WhenTheUserMakesARequestToUpdateTheSupplierWithOrderId(string orderId, Table table)
+        {
+            var supplierTable = table.CreateInstance<SupplierSectionTable>();
+            var addressTable = _context["SupplierAddress"];
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:107]: We should check if these keys exist, and throw a nice helpful error if they don't, so the person using the step knows what went wrong

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs [hunk 5]
```diff
@@ -86,16 +90,75 @@ public async Task ThenTheResponseContainsTheFollowingSupplierAddress(Table table
...
+            var addressId = (int)(await OrderEntity.FetchOrderByOrderId(_settings.ConnectionString, orderId))
+                .SupplierAddressId;
+
+            var actual = await AddressEntity.FetchAddressById(_settings.ConnectionString, addressId);
+
+            actual.Should().BeEquivalentTo(address);
+        }
+
+        [Then(@"the supplier contact for order (.*) is")]
+        public async Task ThenTheSupplierContactIdContactForOrderCIs(string orderId, Table table)
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:141]: I feel like that C at the end of the function name can go

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs [hunk 6]
```diff
@@ -86,16 +90,75 @@ public async Task ThenTheResponseContainsTheFollowingSupplierAddress(Table table
...
+
+            var contactId = (int)(await OrderEntity.FetchOrderByOrderId(_settings.ConnectionString, orderId))
+                .SupplierContactId;
+
+            var actual = await ContactEntity.FetchContactById(_settings.ConnectionString, contactId);
+            actual.Should().BeEquivalentTo(contact);
+        }
+
+        [Then(@"the supplier for order (.*) is updated")]
+        public async Task ThenTheSupplierForOrderCIsUpdated(string orderId, Table table)
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:153]: C's, C's everywhere!

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 7]
```diff
@@ -89,13 +89,17 @@ public static async Task<OrderEntity> FetchOrderByOrderId(string connectionStrin
...
+                          SupplierAddressId,
+                          SupplierContactId,
                           LastUpdated,
                           LastUpdatedBy,
                           LastUpdatedByName,
                           SupplierId,
                           SupplierName
                          FROM dbo.[Order]
-                         WHERE OrderId = @orderId;", new { orderId }));
+                         WHERE OrderId = @orderId;", new {orderId}));
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:102]: I prefer this with the white space, but if it ain't in the standards I can't complain!

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature (hunk 2 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature:22]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/ContactSteps.cs (hunk 3 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/ContactSteps.cs:69]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs (hunk 5 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:141]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs (hunk 6 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:153]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs (hunk 7 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:102]: Nah I do too, dont know why it changed it

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/SupplierSectionController.cs (hunk 1 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/SupplierSectionController.cs:82]: 👍

**Pull request merged**